ui/src/router/components/form/validation/schema.js

```diff
@@ -58,6 +58,60 @@ const validateRuleNames = function(items) {
   return !!errors.length ? new yup.ValidationError(errors) : true;
 };
 
+const validateRoutes = function (items) {
+  const defaultTrafficRule = this.options.parent.default_traffic_rule;
+  let trafficRuleRoutes = [...new Set(this.options.parent.rules.map(rule => rule.routes).flat(1))];
+  if (defaultTrafficRule) {
+    trafficRuleRoutes = [...trafficRuleRoutes, ...defaultTrafficRule.routes]
+  }
+
+  const errors = [];
+  items.forEach((item, idx) => {
+    if (!trafficRuleRoutes.includes(item.id)) {
+      errors.push(
+        this.createError({
+          path: `${this.path}[${idx}].id`,
+          message: "This route should be removed since they have no Traffic Rule(s) associated and will never be called.",
+        })
+      );
+    }
+  });
+
+  // Validate Nop/Standard Ensembler
+  const ensembler_type = this.options.parent.ensembler.type
```

Comment:
I think it's better for the yup UI validation to be warning on the ensembler config rather than the routes for the same reason as above. Besides, we already have the `validRouteSchema` on `fallback_response_route_id` and `final_response_route_id` - can we look at manipulating that definition rather than adding all the ensembler based validation here? We can rename this function from `validateRoutes` to `validateDanglingRoutes`.

Reply:
`validRouteSchema` is currently being used for Custom and Default Traffic Rules schema, and they don't require such sophisticated validation. Perhaps another separate function?

Reply:
Ahh got it. It's related to this: https://github.com/caraml-dev/turing/pull/240/files#r970360776

With the previous workflow, no way you can select an invalid route (previously, invalid route was any route with traffic rules) as final / fallback route. Now the condition for what is valid and invalid has changed. But if we keep that workflow, the UI will not allow such a value. So there is nothing additional to validate here in terms of the traffic rule and default route - such checks are done in the API and this invalid scenario can't be hit from the UI.

No harm having another layer of yup validations but I guess it's not strictly necessary.

Reply:
Guess this PR could have been much simpler 😅 I'll remove this validation then!